mkosi/distributions/gentoo.py

```diff
@@ -295,41 +214,64 @@ def mkosi_conf(self) -> None:
             )
         )
 
-    def get_snapshot_of_portage_tree(self) -> None:
-        run_workspace_command(self.state.root, ["/usr/bin/emerge-webrsync"], network=True,
-                              env=self.state.environment)
-
-    def update_stage3(self) -> None:
-        invoke_emerge(self.state, opts=self.EMERGE_UPDATE_OPTS, pkgs=self.pkgs['boot'])
-        invoke_emerge(self.state, opts=["--config"], pkgs=["sys-kernel/gentoo-kernel-bin"])
-        invoke_emerge(self.state, opts=self.EMERGE_UPDATE_OPTS, pkgs=self.pkgs['sys'])
+        repos_cfg = cls.stage3_cache / BINREPOS_CONF_FILE
+        with repos_cfg.open(mode='a') as f:
+            for repo in state.config.repositories:
+                f.write(
+                    dedent(
+                        f"""\
+                        # MKOSI
+                        [binhost]
+                        sync-uri = {repo}
+                        """
+                    )
+                )
```

Comment:
`--repositories` should only be used to enable repositories that are disabled by default, not to add new ones. For that, we now have `--package-manager-tree` so this should be reworked to pick up `etc/portage/binrepos.conf` from `state.pkgmngr` if it exists there.

Reply:
i have related question. `install_package_manager_trees()` is called before `install_distribution()` in `build_image()`.

look at these two lines in `gentoo.py`:
```
copy_path(config, root_portage_cfg)
copy_path(config, cls.stage3_cache / "etc/portage")
```

am i replicating what `install_package_manager_trees()` does there?

Reply:
see the latest:
1. we append(?) to our defaults whatever user puts in `pkgmngr`
2. we honor `--repositories` in `invoke_emerge()` but we don't document it; i.e. clobber the `pkgmngr` or our `stage3_cache`.